build.go

```diff
@@ -85,85 +79,139 @@ func buildTestCommandLine(exe string, pkg string, tags ...string) []string {
 	if mg.Verbose() {
 		args = append(args, "-v")
 	}
-	args = append(args, formatTags(tags)...)
+	args = append(args, formatTags(goTagOpt, tags)...)
 	return append(args, pkg)
 }
 
-type testBuilder struct {
+// TestBuilder implements [mg.Fn] to build (but not run) the test binary for a single package.
+type TestBuilder struct {
 	pkg  string
 	tags []string
 }
 
-func (tb *testBuilder) Name() string {
+// Name implements [mg.Fn].
+func (tb *TestBuilder) Name() string {
 	return tb.ID()
 }
 
-func (tb *testBuilder) ID() string {
+// ID implements [mg.Fn].
+func (tb *TestBuilder) ID() string {
 	return fmt.Sprintf("build-test-%s", tb.pkg)
 }
 
-func (tb *testBuilder) Run(ctx context.Context) error {
-	return BuildTest(ctx, tb.pkg, tb.tags...)
-}
-
-// BuildTestDep returns a [mg.Fn] that will build the tests for the given package, subject to any given build tags.
-func BuildTestDep(pkg string, tags ...string) mg.Fn {
-	return &testBuilder{pkg, tags}
-}
-
-// BuildTest builds the specified package's test.
-func BuildTest(ctx context.Context, pkg string, tags ...string) error {
+// Run implements [mg.Fn]. If the test binary for the package needs building, then it gets built using the configured
+// build tags, outputting <package-name>.test in the package director.
+func (tb *TestBuilder) Run(ctx context.Context) error {
 	mg.CtxDeps(ctx, LoadDependencies)
-	deps := GetDependencies(pkg, Package.TestFiles, Package.TestImportPackages)
+	deps := GetDependencies(tb.pkg, Package.TestFiles, Package.TestImportPackages)
 	if len(deps) == 0 {
 		return nil
 	}
 
-	info := Packages[pkg]
-	exe := filepath.Join(info.Dir, info.Name+".test")
+	info := Packages[tb.pkg]
+	exe := info.TestBinary()
 
 	newer, err := target.Path(exe, deps...)
 	if err != nil || !newer {
 		return err
 	}
-	return sh.RunV(
-		mg.GoCmd(),
-		buildTestCommandLine(exe, pkg, tags...)...,
-	)
+	return sh.RunV(mg.GoCmd(), buildTestCommandLine(exe, tb.pkg, tb.tags...)...)
 }
 
-type allTestBuilder struct {
-	tags []string
+// BuildTest returns a [mg.Fn] that will build the tests for the given package, subject to any given build tags.
+func BuildTest(pkg string, tags ...string) *TestBuilder {
```

Comment:
This and `TestBuilder` probably don't need to be published. I don't foresee much building or running of individual tests.